### PR TITLE
[8.0-preview2] Only display proxy url in endpoints if service is in ready state

### LINKED ISSUE #1364: [WebToolsE2E] Do Ctrl+F5 of Aspire Starter application a second time and navigate to webfrontend gives 404 page not found error
**REGRESSION INFO**: Works fine with previous build Aspire 8.0.0-preview.2.23610.1

**INSTALL STEPS**
1. Clean machine: Win11 x64 23h2 ENU
2. Install latest VS 17.9 P2 build from [d17.9-34407.89]
3. Install Docker Desktop for Windows
4. CLI workload update and install aspire workload: [version 8.0.0-preview.2.23612.3]
5. Apply NuGet Feeds

**REPRO STEPS**
1. New Aspire Starter Application > AspireStarterApp1
2. Open AspireApp1.AppHost > Program.cs
3. Ctrl+F5 the first time
    - a. Dashboard comes up successfully
    - b. Click the endpoint for webfrontend (eg: localhost:5036 in screenshot below)
    - image
    - c. First-time, the webfrontend page loads successfully
    - image
    - d. Close the browser (both webfrontend and dashboard tabs)
    - e. Close the VS Debug console
4. Ctrl+F5 again
    - a. Dashboard comes up successfully
    - b. Click the endpoint for webfrontend (eg: localhost:5036 in screenshot below)
    - image

**ACTUAL**
Endpoint URL of webfrontend opens with '404 page not found' error
image

**Workaround**: Keep trying to hit the endpoint URL from the dashboard. If we wait about a **_minute or two_**, eventually the page loads successfully.

**EXPECTED**
Endpoint URL shows the webfrontend page successfully

### PR DESCRIPTION
**Pull request by smitpatel**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1405)

### CODE REVIEW

**Comment by JamesNK** [comment]
How come? Can't consumers of the model decide whether to use the endpoints or not based on whether it is ready?

File: src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs [hunk 1]
```diff
@@ -402,7 +402,8 @@ private ProjectViewModel ConvertToProjectViewModel(Executable executable)
             }
 
             var matchingService = _servicesMap.Values.SingleOrDefault(s => s.Metadata.Name == endpoint.Spec.ServiceName);
-            if (matchingService?.UsesHttpProtocol(out var uriScheme) == true)
+            if (string.Equals(matchingService?.Status?.State, "Ready", StringComparison.OrdinalIgnoreCase)
```
Comment (JamesNK) [inline src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs:405]: I want to improve this kind of test. See https://github.com/dotnet/aspire/issues/1406. Just an FYI, it doesn't have to happen in this PR.

**Comment by JamesNK** [comment]
If the goal is to not display endpoint URLs when a resource is not ready, that check should happen in the resources page UI.

Other logic uses the endpoints now. For example, the telemetry pages match outbound addresses with endpoint addresses. If a resource temporarily goes down and it is updated to have no endpoints, then telemetry pages will be unable to match the outbound address to a resource anymore.

File: src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs (hunk 1 above)
Comment (drewnoakes) [reply src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs:405]: Also, to add `StringComparers.ResourceState` so we use the same comparer across values.

**Review by JamesNK — DISMISSED** [review]
See https://github.com/dotnet/aspire/pull/1405#issuecomment-1857185006

**Comment by smitpatel** [comment]
This won't resolve the issue.

**Pull request closed without merging**